base/file.jl

```diff
@@ -424,7 +424,7 @@ function tempdir()
     temppath = Vector{UInt16}(undef, 32767)
     lentemppath = ccall(:GetTempPathW, stdcall, UInt32, (UInt32, Ptr{UInt16}), length(temppath), temppath)
     windowserror("GetTempPath", lentemppath >= length(temppath) || lentemppath == 0)
-    resize!(temppath, lentemppath)
+    temppath[lentemppath] == 0x005c ?  resize!(temppath,lentemppath-1) :  resize!(temppath,lentemppath)
```

Comment:
This would fail if `temppath` were the root directory, e.g. `C:\`, but I guess that shouldn't be possible.

Reply:
Good point, this would occur if `TMP` env variable was set to `C:\\`

Reply:
Latest commit checks that the folder is not a root directory and if it is a root directory does not strip the trailing slash

Reply:
This fixes the case where `TMP` is set to `C:\\`  (kinda pathological, but in theory possible)